build.gradle

```diff
@@ -66,54 +77,42 @@ def skippedTaskNames = System.getenv().findAll { key, value ->
 
 gradle.startParameter.excludedTaskNames += skippedTaskNames
 
-checkstyle
-{
-    toolVersion = versions.checkstyle
-}
-
-test
-{
-    testLogging
-    {
+test {
+    testLogging {
         events "passed", "skipped", "failed"
     }
 }
 
-configurations
-{
-    all
-    {
-        resolutionStrategy
-        {
-            force packages.atlas
-        }
-        exclude group: 'org.slf4j', module: 'slf4j-log4j12'
-    }
-}
-
-dependencies
-{
-    packIntoJar packages.atlas
-}
-
 tasks.withType(Test) {
-    reports.html.destination = file("${reporting.baseDir}/${name}")
+    // Use `build/reports/$testTaskName` instead of `build/reports/$sourceSetName`
+    reports.html.setDestination(file("${reporting.baseDir}/${name}"))
 }
 
-// Javadoc currently fails on a dependency issue
-//task javadocJar(type: Jar) {
-//    classifier = 'javadoc'
-//    from javadoc
-//}
+task javadocJar(type: Jar) {
+    classifier = 'javadoc'
+    from javadoc
+}
 
 task sourcesJar(type: Jar) {
     classifier = 'sources'
     from sourceSets.main.allSource
 }
 
-artifacts
-{
-    archives /*javadocJar,*/ sourcesJar
+artifacts {
+    archives javadocJar, sourcesJar
+}
+
+if (System.env.containsKey("GPG_KEY_ID") && System.env.containsKey("GPG_PASSPHRASE") && System.env.containsKey("GPG_KEY_LOCATION")) {
+    ext."signing.keyId" = System.env.GPG_KEY_ID
+    ext."signing.password" = System.env.GPG_PASSPHRASE
+    ext."signing.secretKeyRingFile" = System.env.GPG_KEY_LOCATION
+}
+if (!ext.has("signing.keyId") || !ext.has("signing.password") || !ext.has("signing.secretKeyRingFile")) {
```

Comment:
Is it possible to fail the build in that case? It would be better to have the artifacts signed always.

Reply:
It's possible, but builds that are not running on Travis should still be possible without signing setup. I'll add a condition for that.

Reply:
I added 0c58af8d0116f611b1b26ef235ff7ae0bb0ce808 below to address this.
The build now fails if all of the following conditions are met:
* the environment variable `TRAVIS_REPO_SLUG` is set to `osmlab/josm-atlas`
* the environment variable `TRAVIS_PULL_REQUEST` is set to `false` (PRs can't access encrypted variables for security reasons)
* the signing options are not set up